plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/RestCatalogTestUtils.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.rest;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.trino.hdfs.DynamicHdfsConfiguration;
+import io.trino.hdfs.HdfsConfig;
+import io.trino.hdfs.HdfsConfiguration;
+import io.trino.hdfs.HdfsConfigurationInitializer;
+import io.trino.hdfs.HdfsContext;
+import io.trino.hdfs.HdfsEnvironment;
+import io.trino.hdfs.authentication.NoHdfsAuthentication;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.testing.TestingConnectorSession;
+import org.apache.hadoop.fs.Path;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.jdbc.JdbcCatalog;
+import org.assertj.core.util.Files;
+
+import java.io.File;
+
+public final class RestCatalogTestUtils
+{
+    private RestCatalogTestUtils() {}
+
+    public static Catalog backendCatalog(File warehouseLocation)
+    {
+        ImmutableMap.Builder<String, String> properties = ImmutableMap.builder();
+        properties.put(CatalogProperties.URI, "jdbc:sqlite:file:" + Files.newTemporaryFile().getAbsolutePath() + "_mode=memory?limit_length=4096");
```

Comment:
Can we use H2 for this? This would be the first dependency on SQLite, so best to avoid if possible.

Reply:
All of the existing testing in Iceberg is performed via Sqlite so that seems more likely to be consistent, but I was able to switch it to H2 with only updates required for some of the length tests.  This will remove the additional dependency, but might result in issues in the future.